pkg/boot/universalpayload/handoffblock.go

```diff
@@ -0,0 +1,235 @@
+	EfiResourceFirmwareDevice     = 0x00000003
+	EfiResourceMemoryMappedIOPort = 0x00000004
+	EfiResourceMemoryReserved     = 0x00000005
+	EfiResourceIOReserved         = 0x00000006
+	EfiResourceMemoryUnaccepted   = 0x00000007
+	EfiResourceMaxMemoryType      = 0x00000008
+	EfiResourceUnimplemeted       = 0x00000008
+)
+
+// Efi Resource Attributes
+const (
+	EfiResourceAttributePresent               = 0x00000001
+	EfiResourceAttributeInitialized           = 0x00000002
+	EfiResourceAttributeTested                = 0x00000004
+	EfiResourceAttributeReadProtected         = 0x00000080
+	EfiResourceAttributeWriteProtected        = 0x00000100
+	EfiResourceAttributeExecutionProtected    = 0x00000200
+	EfiResourceAttributePersistent            = 0x00800000
+	EfiResourceAttributeSingleBitECC          = 0x00000008
+	EfiResourceAttributeMultipleBitECC        = 0x00000010
+	EfiResourceAttributeECCReserved1          = 0x00000020
+	EfiResourceAttributeECCReserved2          = 0x00000040
+	EfiResourceAttributeUncacheable           = 0x00000400
+	EfiResourceAttributeWriteCombineable      = 0x00000800
+	EfiResourceAttributeWriteThroughCacheable = 0x00001000
+	EfiResourceAttributeWriteBackCacheable    = 0x00002000
+	EfiResourceAttribute16BitIO               = 0x00004000
+	EfiResourceAttribute32BitIO               = 0x00008000
+	EfiResourceAttribute64BitIO               = 0x00010000
+	EfiResourceAttributeUncachedExported      = 0x00020000
+	EfiResourceAttributeReadProtectable       = 0x00100000
+	EfiResourceAttributeWriteProtectable      = 0x00200000
+	EfiResourceAttributeExecutionProtectable  = 0x00400000
+	EfiResourceAttributePersistable           = 0x01000000
+	EfiResourceAttributeReadOnlyProtected     = 0x00040000
+	EfiResourceAttributeReadOnlyProtectable   = 0x00080000
+	EfiResourceAttributeEncrypted             = 0x04000000
+	EfiResourceAttributeSpecialPurpose        = 0x08000000
+	EfiResourceAttributeMoreReliable          = 0x02000000
+)
+
+// HandoffHob values
+const (
+	EfiHobHandoffInfoVersion          = 0x09
+	EfiHobHandoffInfoBootMode         = 0x0
+	EfiHobHandoffInfoEfiMemoryTop     = 0x4E00000
+	EfiHobHandoffInfoEfiMemoryBottom  = 0x800000
+	EfiHobHandoffInfoFreeEfiMemoryTop = 0x4E00000
+	EfiHobHandoffInfoFreeMemoryBottom = 0xE00000
+)
+
+// According to Intel SDM Volume 1 Chapter 19.3 "I/O ADDRESS SPACE":
+// The I/O address space consists of 2^16 (64K) individually addressable
+// 8-bit I/O ports, numbered 0 through FFFFH.
+// Set the default IO Address size to 16
+const (
+	DefaultIOAddressSize = 16
+)
+
+// Type Aliases
+type EfiBootMode uint32
+type EfiResourceType uint32
+type EfiPhysicalAddress uint64
+type EfiResourceAttributeType uint32
+
+// Efi HOB Generic Header
+type EfiHobGenericHeader struct {
+	HobType   uint16
+	HobLength uint16
+	Reserved  uint32
+}
+
+type EfiHobHandoffInfoTable struct {
+	Header              EfiHobGenericHeader
+	Version             uint32
+	BootMode            EfiBootMode
+	EfiMemoryTop        EfiPhysicalAddress
```

Comment:
So you really need EfiMemoryTop or could it be MemoryTop

Reply:
`EfiMemoryTop` is preferred according to spec: https://www.intel.com/content/dam/doc/reference-guide/efi-hand-off-block-specification.pdf  (#page 18, see `EFI_HOB_HANDOFF_INFO_TABLE` )